agent/lib/kontena/service_pods/creator.rb

```diff
@@ -78,21 +80,43 @@ def perform
         service_container
       end
 
-      # @param [Docker::Container] service_container
-      # @param [String] type
       # @return [Boolean]
-      def run_hooks(service_container, type)
-        service_pod.hooks.each do |hook|
-          if hook['type'] == type
-            info "running #{type} hook: #{hook['cmd']}"
-            command = ['/bin/sh', '-c', hook['cmd']]
-            log_hook_output(service_container.id, ["running #{type} hook: #{hook['cmd']}"], 'stdout')
-            stdout, stderr, exit_code = service_container.exec(command)
-            log_hook_output(service_container.id, stdout, 'stdout')
-            log_hook_output(service_container.id, stderr, 'stderr')
-            if exit_code != 0
-              raise "Failed to execute hook: #{hook['cmd']}"
+      def run_pre_start_hooks
+        hooks_for('pre_start').each do |hook|
+          service_config = config_container(service_pod.dup)
+          service_config['Cmd'] = ['/bin/sh', '-c', hook['cmd']]
```

Comment:
Should this override the entrypoint? or perhaps override that entrypoint is `['/bin/sh', '-c']` and then put the cmd in place. I think users will expect to execute script like things with hooks and now if my entrypoint is some binary then these will most likely fail.

Also could there be only one temp container for all the pre-hooks? That would mean to modify the entrypoint to be some "pause" thing and then all the hooks would be `exec`'s.